lib/services/volunteer.js

```diff
@@ -2,137 +2,80 @@ const Schmervice = require('schmervice');
-          volunteer.lang = null;
-          volunteer.engagement_in_months = 0;
-        }
+      volunteers.forEach((v, i) => {
+        volunteers[i].classes = classes_data[i];
       });
 
-      volunteers_data.count = volunteer_ids.length;
-      volunteers_data.volunteers = volunteers;
-    } catch (err) {
-      return [err, errorHandler(err)];
-    }
-    return [null, volunteers_data];
-  }
-
-  async getVolunteerById(id) {
-    const { Classes, ClassRegistrations } = this.server.models();
-    let volunteer_data = {};
-
-    try {
-      // get classes details
-      const classes_data = await Classes.query().where('facilitator_id', id);
-
-      if (classes_data.length > 0) {
-        volunteer_data = {
-          facilitator_id: classes_data[0].facilitator_id,
-          facilitator_name: classes_data[0].facilitator_name,
-          facilitator_email: classes_data[0].facilitator_email,
-        };
-      }
-
-      const ratings = await Promise.all(
-        classes_data.map((c) => {
-          return ClassRegistrations.query()
-            .select('feedback')
-            .where('class_id', c.id)
-            .whereNotNull('feedback');
-        })
-      );
-
-      // filter classes data and get ratings
-      ratings.forEach((c, i) => {
-        c.forEach((r, j) => {
-          ratings[i][j] = Number(r.feedback);
+      // filtering data
+      const res = volunteers.map((v) => {
+        // filter pathways
+        const pathways = [];
+        const pathways_data = v.classes.map((cl) => {
+          return cl.pathways.map((p) => {
+            return p.pathway.name;
+          });
         });
-        classes_data[i] = {
-          id: classes_data[i].id,
-          title: classes_data[i].title,
-          start_time: classes_data[i].start_time,
-          end_time: classes_data[i].end_time,
-          lang: classes_data[i].lang,
-          max_enrollment: classes_data[i].max_enrolment,
-          avg_rating: Math.round(ratings.reduce((a, b) => a + b) / ratings.length),
+        pathways_data.forEach((p) => {
+          p.forEach((s) => {
+            pathways.push(s);
+          });
+        });
+        const uniq_pathways = [...new Set(pathways)];
+
+        return {
+          id: v.id,
+          name: v.name,
+          email: v.email,
+          partner_id: v.partner_id,
+          partner: v.partner ? v.partner.name : '',
+          pathways: uniq_pathways,
+          classes: v.classes.map((cl) => {
+            return {
+              id: cl.id,
+              title: cl.title,
+              description: cl.description,
+              start_time: cl.start_time,
+              end_time: cl.end_time,
+              course_id: cl.course_id,
+              lang: cl.lang,
+              max_enrollment: cl.max_enrolment,
+              recurring_id: cl.recurring_id,
+              ratings: cl.feedbacks.map((r) => {
+                return {
+                  rating: r.feedback,
+                };
+              }),
```

Comment:
Can write in one line if you wrap the object in parentheses:
```suggestion
              ratings: cl.feedbacks.map((r) => ({rating: r.feedback})),
```